lib/utils/extension/src/stream.dart

```diff
@@ -5,6 +5,32 @@ extension StreamExtensionWhereNotNull<T> on Stream<T?> {
 }
 
 extension StreamExtension<T> on Stream<T> {
+  StreamSubscription<T> asyncDropListen<E>(
+      Future<void> Function(T event) convert) {
+    T? lastEvent;
+    var running = false;
+    final lock = Lock();
+
+    return listen((event) async {
+      if (running) {
+        lastEvent = event;
+        return;
+      }
+      // ensure only one event is processed at a time
+      await lock.synchronized(() async {
+        running = true;
+        await convert(event);
+        while (true) {
+          final e = lastEvent;
+          if (e == null) break;
+          await convert(e);
+          lastEvent = null;
```

Comment:
this looks like will break the loop, so this loop will only run once in an event callback whatever lastEvent is. Does this is the  expected behavior?